.github/scripts/ci_failure_bot.py

```
@@ -0,0 +1,312 @@
+                    # Also get step details
+                    for step in job.steps:
+                        if step.conclusion == "failure":
+                            build_logs.append(
+                                {
+                                    "job_name": job.name,
+                                    "step_name": step.name,
+                                    "step_number": step.number,
+                                }
+                            )
+
+            return build_logs
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)